### PR TITLE
TT-12021 Deprecate dashboard.hashKeys

### PR DESCRIPTION
**Pull request by caroltyk**
## Description

dashboard.hashKeys is deprecated now. Users should use `global.hashKeys` field to set key hashing instead.

The global.hashKeys will help to ensure configuration alignment between gateway, dashboard, and MDCB components.

## Test Coverage For This Change

dashboard.hashKeys is deprecated so setting it will not affect the hashKeys configuration. Dashboard will take the hashkey configuration value from global.hashKeys field only.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by caroltyk** [comment]
@buraksekili kindly help to verify if this change is correct.

File: tyk-control-plane/values.yaml [hunk 1]
```diff
@@ -228,10 +228,11 @@ global:
     # It is used to set TYK_MDCB_SECURITY_PRIVATECERTIFICATEENCODINGSECRET
     privateCertificateEncodingSecret: ""
 
-  # hashKeys specifies that if your Tyk Gateway is using hashed keys, set this value to true, so it matches.
-  # The Dashboard will now operate in a mode that is compatible with key hashing.
-  # It will enable hash keys for Tyk Gateway
-  # It is used to set TYK_GW_HASHKEYS
+  # hashKeys specifies that if your Tyk Gateway is using hashed keys, set this value to true so it matches.
+  # The Dashboard and MDCB will now operate in a mode that is compatible with key hashing.
+  # This takes precedence over the dashboard value i.e `dashboard.hashKeys`
```
Comment (buraksekili) [inline tyk-control-plane/values.yaml:233]: At the moment, `dashboard.hashKeys` takes precedence until we deprecate it. So, this comment needs to be updated accordingly. Or, we should update precedence order of hashKeys.

File: tyk-control-plane/values.yaml [hunk 2]
```diff
@@ -228,10 +228,11 @@ global:
...
+  # It is used to set TYK_DB_HASHKEYS and TYK_MDCB_HASHKEYS
+  # It will enable hashkeys in both mdcb and dashboard.
```
Comment (buraksekili) [inline tyk-control-plane/values.yaml:235]: `.global.hashKeys ` also applies to Tyk Gateway as well. So, it also configures `TYK_GW_HASHKEYS`

Comment (caroltyk) [reply tyk-control-plane/values.yaml]: Updated

File: tyk-control-plane/values.yaml (hunk 1 above)
Comment (buraksekili) [reply tyk-control-plane/values.yaml]: looks good, thank you! also, as it is deprecated, we should mention it in the release notes and maybe remove the deprecated field in the future releases

**Review by buraksekili — APPROVED** [review]

File: tyk-control-plane/values.yaml (hunk 1 above)
Comment (caroltyk) [reply tyk-control-plane/values.yaml]: Thanks. Added this to https://github.com/TykTechnologies/tyk-docs/pull/4495 description as a reminder.

**Pull request merged**